src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -23,25 +23,34 @@ internal sealed class DeclarationTreeBuilder : CSharpSyntaxVisitor<SingleNamespa
         private readonly string _scriptClassName;
         private readonly bool _isSubmission;
 
+        // Cleared once we hit the first type.
+        private ImmutableSegmentedHashSet<string> _previousMemberNames;
```

Comment:
>private [](http://example.com/codeflow?start=8&length=7)

readonly?

Reply:
Not in this case.  It's cleared out after we use it hte first time.  Docs in the state object explain why.